internal/db/queryBuilder.go

```diff
@@ -81,25 +81,24 @@ func (qb *queryBuilder) updatedObjectsCTE() string {
 		ignoresPredicate = "AND o.path NOT LIKE ALL(__ignores__::text[])"
 	}
 
-	return fmt.Sprintf(template, bytesSelector, contentsPredicate, pathPredicate, ignoresPredicate)
+	hashSelector := "null::hash as hash"
+	if qb.objectQuery.WithHash {
+		hashSelector = "o.hash"
+	}
+
+	return fmt.Sprintf(template, bytesSelector, hashSelector, contentsPredicate, pathPredicate, ignoresPredicate)
 }
 
-func (qb *queryBuilder) removedObjectsCTE() string {
+func (qb *queryBuilder) removedObjectsCTE(skipChanged bool) string {
```

Comment:
So I changed this because I wanted to be able to delete files which have changed and replace them with their new versions. This is 2 steps where we set a stop version on the removed files and then an insert on all changed files. If we filtered out changed files we would have to intersect the existing files in target with changed files in source